examples/arithmetics/src/language-server/arithmetics-validator.ts

```diff
@@ -9,15 +9,50 @@ export class ArithmeticsValidationRegistry extends ValidationRegistry {
         super(services);
         const validator = services.validation.ArithmeticsValidator;
         const checks: LangiumGrammarChecks = {
-            Division: validator.checkDivByZero
+            Division: validator.checkDivByZero,
+            Definition: validator.checkNormalisable
         };
         this.register(checks, validator);
     }
 }
 export class ArithmeticsValidator {
     checkDivByZero(div: Division, accept: ValidationAcceptor): void {
-        if (isNumberLiteral(div.right) && Number((div.right as NumberLiteral).value) === 0) {
+        if (isNumberLiteral(div.right) && +div.right.value === 0) {
             accept('error', 'Division by zero is detected.', { node: div, property: 'right' });
         }
     }
+
+    checkNormalisable(def: Definition, accept: ValidationAcceptor): void {
+        const context = new Map<Expression, number>();
+
+        const makeOp = (expr: Addition | Subtraction | Multiplication | Division, op: (x: number, y: number) => number): void => {
+            const subExprs = [expr.left, expr.right];
+            subExprs.forEach(e => evalExpr(e));
+            const [left, right] = subExprs.map(e => isNumberLiteral(e) ? +e.value : context.get(e));
```

Comment:
Is there a specified reason why you start the `e.value` statement with a +?

Reply:
To apply `op`, we have to have `left` and `right` of type `number`. So, I just convert `string` of `e.value` to `number`.